opsdroid/connector/slack/__init__.py

```diff
@@ -265,7 +265,19 @@ def __init__(self, config, opsdroid=None):
                         target=payload["channel"]["id"],
                         connector=self,
                     )
-                    await block_action.update_entity("value", action["value"])
+
+                    action_value = None
+                    if action["type"] == "button":
+                        action_value = action["value"]
+                    elif action["type"] in ["overflow", "static_select"]:
+                        action_value = action["selected_option"]["value"]
+                    elif action["type"] == "datepicker":
+                        action_value = action["selected_date"]
+                    elif action["type"] == "multi_static_select":
+                        action_value = [v["value"] for v in action["selected_options"]]
```

Comment:
@jacobtomlinson , I'm not sure about this line and I'd appreciate your input -
Currently, setting an entity (via `Event.update_entity(..)`) to a `list` won't always work with the code in [event_type.py:36](https://github.com/opsdroid/opsdroid/blob/3a265fd76aa0d7055d60329bc244ca075fc1761e/opsdroid/parsers/event_type.py#L36), because if the user will try to filter using a list which is not ordered the same as the list we got from slack that comparison will not work.

I can change the code in event_type.py to handle different types of entities correctly but I'm not entirely sure how it will impact other parts of opsdroid which I'm not entirely familiar with (yet).

What do you think?

Reply:
I think it would be totally reasonable to check if the entity is a list and if so sort it before comparing.